sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/model/RowWrapper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.stargate.sgv2.docsapi.model;
+
+import io.stargate.bridge.grpc.Values;
+import io.stargate.bridge.proto.QueryOuterClass.ColumnSpec;
+import io.stargate.bridge.proto.QueryOuterClass.Row;
+import io.stargate.bridge.proto.QueryOuterClass.Value;
+import java.util.List;
+import java.util.Objects;
+
+/** A wrapper around the protobuf-generated {@link Row}, to add the column metadata. */
+public class RowWrapper {
```

Comment:
* this could be immutable
* we could port this here as lazy evaluated prop: https://github.com/stargate/stargate/pull/1797/files#diff-073be9eaa3353270c2e732f3821a3d15a0f721ad06ba57b7d1797408c37dea92R59-R66